binary.json

```diff
@@ -46,14 +46,24 @@
          "expected": 31
       },
       {
-         "description": "numbers other than one and zero raise an error",
-         "binary": ["012", "2"],
-         "expected": -1
+        "description": "2 is not a valid binary digit and raises an error",
+        "binary": "2",
+        "expected": -1
       },
       {
-         "description": "containing letters raises an error",
-         "binary": ["10nope", "nope10", "10nope10", "001 nope"],
```

Comment:
The distinct cases we have here are:
* trailing non-binary characters
* leading non-binary characters
* internal non-binary characters
* a number and a word whitespace separated
They're probably all good to include.